File: 240803/나누고 빼면서 합하기/divide-and-subtract-and-add-up.py
n, m = map(int,input().split())
_list = list(map(int, input().split()))

def find_ans(_list, m):
    _sum = 0

    while m >= 1:
        _sum += _list[m-1]

        if m % 2:
            m -= 1
        else:
            m //= 2

    return _sum

print(find_ans(_list, m))

File: 240809/나누고 빼면서 합하기/divide-and-subtract-and-add-up.js
const fs = require('fs');
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const numbers = input[1].split(' ').map(Number)

let tempIndex = m;

let sumVal = numbers[m-1];
while ( tempIndex > 1 ) {
    if ( tempIndex % 2 === 1 ) {
        tempIndex -= 1;
        sumVal += numbers[tempIndex-1];
    } else {
        tempIndex /= 2;
        sumVal += numbers[tempIndex-1]
    }
}

console.log(sumVal)